### PR TITLE
Added golint / govet to Travis CI for arm packages

### PR DESCRIPTION
**Pull request by brendandixon**
This change:

- Improves the UserAgent string to disambiguate arm packages from others in the SDK
- Improves setting the http.Response into generated results (reduces likelihood of a nil reference)
- Adds gofmt, golint, and govet to Travis CI for the arm packages

Signed-off-by: Brendan Dixon <brendand@microsoft.com>

### CODE REVIEW

**Comment by brendandixon** [comment]
This PR address #196.

File: arm/subscriptions/version.go [hunk 1]
```diff
@@ -25,11 +25,11 @@ import (
 const (
 	major = "0"
 	minor = "1"
-	patch = "0"
+	patch = "1"
```
Comment (ahmetalpbalkan) [inline arm/subscriptions/version.go]: this means you need to update the changelog?

File: .travis.yml [hunk 2]
```diff
@@ -8,9 +8,12 @@ before_script:
 
 go: tip
 script:
+  - test -z "$(gofmt -s -l $(find ./arm/* -type d -print) | tee /dev/stderr)"
   - test -z "$(gofmt -s -l -w management | tee /dev/stderr)"
   - test -z "$(gofmt -s -l -w storage | tee /dev/stderr)"
   - go build -v ./...
+  - test -z "$(go vet $(find ./arm/* -type d -print) | tee /dev/stderr)"
```
Comment (ahmetalpbalkan) [inline .travis.yml]: Is there a shorter form of `go vet` that doesn't require you to do `$(find ...)`? Maybe like  `go vet -v ./arm` or `go vet -v ./arm/...` ?

Comment (brendandixon) [inline .travis.yml]: @ahmetalpbalkan Not that I could find. I tried various incantations before settling on this. I felt it poor form to list each package individually since it would mean having to update the file any time we added a new package.

File: arm/subscriptions/version.go (hunk 1 above)
Comment (brendandixon) [inline arm/subscriptions/version.go]: Grrr. Correct.

Comment (ahmetalpbalkan) [inline arm/subscriptions/version.go]: if you haven't tagged you probably shouldn't bump the version up. just undo the change in `patch`.

**Comment by ahmetalpbalkan** [comment]
LGTM
